torch/autograd/profiler.py

```diff
@@ -308,6 +308,47 @@ def self_cpu_time_total(self):
         return self.function_events.self_cpu_time_total
 
 
+class record_function(object):
+    """Context manager that adds a label to a block of Python code when running autograd
+    profiler.  It is useful when tracing the code profile.
+
+    Arguments:
+        name (str): Label assigned to the block of code.
+
+    Example:
+        >>> x = torch.randn((1, 1), requires_grad=True)
+        >>> with torch.autograd.profiler.profile() as prof:
+        ...     y = x ** 2
+        ...     with torch.autograd.profiler.record_function("label-z"): // label the block
+        ...         z = y ** 3
+        ...     y.backward()
+        ...
+        >>> # NOTE: some columns were removed for brevity
```

Comment:
Side topic, shall we avoid printing those cuda columns when use_cuda is `False`?